crates/flashblocks-rpc/src/rpc.rs

```diff
@@ -126,19 +127,26 @@ impl<E> EthApiExt<E> {
         &self,
         tx: Recovered<OpTransactionSigned>,
         tx_info: TransactionInfo,
+        deposit_receipt: Option<OpDepositReceipt>,
     ) -> Transaction {
         let tx = tx.convert::<OpTxEnvelope>();
         let mut deposit_receipt_version = None;
         let mut deposit_nonce = None;
 
         if tx.is_deposit() {
-            let receipt = self
-                .cache
-                .get::<OpReceipt>(&format!("receipt:{:?}", tx_info.hash.unwrap().to_string()))
-                .unwrap();
-            if let OpReceipt::Deposit(receipt) = receipt {
+            if let Some(receipt) = deposit_receipt {
                 deposit_receipt_version = receipt.deposit_receipt_version;
                 deposit_nonce = receipt.deposit_nonce;
+            } else {
+                let cached_receipt = self
+                    .cache
+                    .get::<OpReceipt>(&format!("receipt:{:?}", tx_info.hash.unwrap().to_string()))
+                    .unwrap();
```

Comment:
Is it possible this cache grab can be null? If it is then this will panic.

Reply:
This is where the panic used to be before the change, hut not anymore. Deposit transactions only have two pathways to this function - historical come in via TransactionSource::Block, which will reach out to the Eth API, or they exist purely in cache and havent been included in a block yet so they will exist in cache.

TransactionSource::Pool isnt a valid pathway for deposits, which would've been a case where this panics, because system transactions dont go in the mempool.